src/choosepsp.ts

```diff
@@ -1,19 +1,74 @@
-import { userSession } from './js/sessiondata';
-import { modalWindows } from './js/modals';
+import * as TE from 'fp-ts/lib/TaskEither';
+// FAKE JSON
+// import psp from './assets/json/psp.json';
+import { Millisecond } from 'italia-ts-commons/lib/units';
+import { toError } from 'fp-ts/lib/Either';
+import { createClient } from '../generated/definitions/pagopa/client';
+import { retryingFetch } from './utils/fetch';
+import idpayguard from './js/idpayguard';
+import { initHeader } from './js/header';
 
-import psp from './assets/json/psp.json';
+const pmClient = createClient({
+  baseUrl: 'http://localhost:8080',
+  fetchApi: retryingFetch(fetch, 2000 as Millisecond, 3),
+});
+
+document.addEventListener('DOMContentLoaded', async () => {
+  // idpayguard
+  idpayguard();
+
+  // initHeader
+  initHeader();
 
-document.addEventListener('DOMContentLoaded', () => {
-  userSession();
+  const walletStored = sessionStorage.getItem('wallet') || '';
+  const checkDataStored = sessionStorage.getItem('checkData') || '';
+  const sessionToken = sessionStorage.getItem('sessionToken') || '';
 
-  // init modals
-  modalWindows();
+  const checkData = JSON.parse(checkDataStored);
+  const wallet = JSON.parse(walletStored);
 
-  const eventList = (el: any) => {
-    positionel?.querySelector('.active')?.classList.remove('active');
+  const idPayment = checkData.idPayment;
+  const Bearer = `Bearer ${sessionToken}`;
+  const paymentType = wallet.type;
+  const isList = true;
+  const language = 'it';
+  const idWallet = wallet.idWallet;
+
+  const psp = await TE.tryCatch(
+    () =>
+      pmClient.getPspListUsingGET({
+        Bearer,
+        paymentType,
+        isList,
+        idWallet,
+        language,
+        idPayment,
+      }),
+    toError,
+  )
+    .fold(
+      () => [], // to be replaced with logic to handle failures
+      myResExt =>
+        myResExt
+          .fold(
+            () => [],
+            myRes => (myRes.status === 200 ? myRes.value.data?.pspList : []),
+          )
+          .map(e => ({
+            name: e?.businessName,
+            label: e?.businessName,
+            image: e?.logoPSP,
+            commission: e?.fixedCost?.amount / Math.pow(10, e?.fixedCost?.decimalDigits),
+          })),
+    )
+    .run();
+
```

Comment:
@vito80ba  @pasqualespica Maybe we should add also the call to `PUT wallet` or do you think we should open another PR?

Reply:
@GiovanniMancini we got a discount, for convenience we put everything here